### DISCUSSION TITLE
Data set for yolov8s which is used in "hornet3000"

### DISCUSSION THREAD

**Discussion by saichaitanya-c** [vespCV/hornet3000#1] [Q&A]
Hello ,
hi where i can find yolov8s dataset which is used in this project([hornet3000](https://github.com/vespCV/hornet3000)
)? i can able to find yolov10n dataset but not for yolov8s . can you please help me in this..if you share that dataset it is very helpful ......

**Comment by marcory-hub** [comment]
Hi @saichaitanya-c , 
Thank you for your interest in the project! The YOLOv8s model was trained on a dataset that is not currently available online. I’m working on finding a free and accessible space to host the dataset. In the meantime, you can use the [Hornet3000 dataset](https://www.kaggle.com/datasets/marcoryvandijk/vespa-velutina-v-crabro-vespulina-vulgaris) to create your own model.

**Comment by saichaitanya-c** [comment]
Hi @marcory-hub Thanks for the response. I am looking for the YOLOv8s model data set which is used in the project. (hornet3000). It would be really helpful if you can help me with it. If possible can you send me dataset to my mail id : kancharanasaichaitanya@gmail.com

 I want to convert that data set onnx to Hef file for Halio 8L AI accelerator. For this, I would need data set. Incase if you have Hef file , Could you Please share the Hef file for the YOLOv8s model. Then i would not require the data set.

**Comment by marcory-hub** [comment]
Hi @saichaitanya-c,

Unfortunately, I got stuck when trying to compile it to an hef file. I tried to run DFC in Colab, but it gives an error in the final compilation step from har to hef.

Today, I will spin up a Google Cloud Engine to try the compilation on that platform. If that doesn't work, I will send you the current dataset and callibration set by mail, so you can experiment with it.

**Comment by saichaitanya-c** [comment]
Thank you so much, @marcory-hub , for your help. I really appreciate your kindness! 😊

**Comment by marcory-hub** [comment]
I'm encountering a bunch of issues compiling the yolov8s model to hef in Colab or Google Cloud. Hoping you can give it a shot with the best.pt and calibration dataset I'm sending over. By the way, the yolov8s model actually runs on a Raspberry Pi 4 4GB (without Hailo) - just using it to test the .pt to hef conversion process in the cloud.

**Comment by saichaitanya-c** [comment]
Hi @marcory-hub ,

I hope you're doing well.

Thank you for sharing your insights and the calibration dataset. After receiving the calibration dataset, I conducted several trials and successfully generated a .hef file. However, upon deploying it on a Raspberry Pi with an AI accelerator, I encountered an issue: every hornet is being detected as either Vespa or Vzon.

I discussed this issue with ChatGPT, and it suggested that the problem might stem from the calibration dataset. Specifically, it highlighted the importance of using the same images for calibration as those used for training the YOLOv8s model with the best.pt file. This mismatch might be causing the incorrect detections.

Could you kindly send me the actual images that were used for training the YOLOv8s model (best.pt)? This dataset would be immensely helpful for my project and might resolve the issue. I have also attached the .hef file I generated for your reference.

Thank you for your assistance.

**Reply by marcory-hub** [reply]
Hi @saichaitanya-c ,

Thanks for all your work, the AI-kit is not the easiest to start with 😅

If you want to create a larger calibration set, the fastest way is likely to use the images from the hornet3000 dataset (they are on kaggle). These images are also included in the larger dataset.

If you only require a functional detector, both the YOLOv8s and YOLOv10n models perform well on a Raspberry Pi 4 with 4GB of RAM (without the need for an AI Kit).

It's interesting that it only detects Vespa and Vzon, not sure if chatGPT is right about the calibrationset (I got not the best answers from LLMs about detailed CV questions). Perhaps you could post a question about this in the Hailo community?
